source/GameData.cpp

```diff
@@ -458,9 +458,14 @@ void GameData::WriteEconomy(DataWriter &out)
 		{
 			out.Write("purchases");
 			out.BeginChild();
-			for(const auto &pit : purchases)
-				for(const auto &cit : pit.second)
-					out.Write(pit.first->Name(), cit.first, cit.second);
+			using Purchase = pair<const System * const, map<string, int>>;
```

Comment:
Other places where you've done this don't have the space here.
```suggestion
			using Purchase = pair<const System *const, map<string, int>>;
```